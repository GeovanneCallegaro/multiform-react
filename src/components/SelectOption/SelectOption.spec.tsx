import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

import { SelectOption } from ".";


describe('<SelectOption />', () => {
  it('should select option is rendered', () => {
    const fn = jest.fn()
    render(<SelectOption
      title = 'title 1'
      description = 'body 1'
      icon = 'icon 1'
      selected = {true}
      handleClick = {fn}
    />)

    expect(screen.getByText('title 1')).toBeInTheDocument();
    expect(screen.getByText('body 1')).toBeInTheDocument();
    expect(screen.getByText('icon 1')).toBeInTheDocument();
  })

  it('should be called the onclick function', () => {
    const fn = jest.fn()
    const {container} = render(<SelectOption
      title = 'title 1'
      description = 'body 1'
      icon = 'icon 1'
      selected = {true}
      handleClick = {fn}
    />);

    userEvent.click(container.firstElementChild);
    expect(fn).toBeCalledTimes(1);
  })

  it('should match snapshot', () => {
    const fn = jest.fn();
    const {container} = render(<SelectOption
      title = 'title 1'
      description = 'body 1'
      icon = 'icon 1'
      selected = {true}
      handleClick = {fn}
    />);

    expect(container.firstElementChild).toMatchSnapshot()
  })
})