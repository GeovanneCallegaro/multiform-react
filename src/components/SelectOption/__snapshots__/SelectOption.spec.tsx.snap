// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectOption /> should match snapshot 1`] = `
<div
  class="sc-bczRLJ bgLQFX containerOption"
>
  <div
    class="sc-gsnTZi dMXCmX"
  >
    icon 1
  </div>
  <div
    class="sc-dkzDqf PTSyZ"
  >
    <div
      class="sc-hKMtZM HetDM"
    >
      title 1
    </div>
    <div
      class="sc-eCYdqJ hIjRhp"
    >
      body 1
    </div>
  </div>
</div>
`;
